src/Orchestration/Adapter.php

```diff
@@ -38,32 +36,68 @@ abstract public function pull(string $image): bool;
     /**
      * List Containers
      *
-     * @return bool
+     * @return array
      */
-    abstract public function list(): bool;
+    abstract public function list(): array;
+
+    /**
+     * Filter ENV vars
+     * 
+     * @param string $string
+     * 
+     * @return string
+     */
+    public function filterEnvKey(string $string): string
+    {
+        $string     = \str_split($string);
+        $output     = '';
+    
+        foreach ($string as $char) {
+            if(\array_key_exists($char, array_fill_keys(\str_split('0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz_'), true))) {
```

Comment:
Lets make this array of chars a value itself, so we don't have to split the string on every iteration.